Src/Core/Primitives/DateTimeAssertions.cs

```diff
@@ -435,5 +435,37 @@ public DateTimeRangeAssertions BeLessThan(TimeSpan timeSpan)
         {
             return new DateTimeRangeAssertions(this, Subject, TimeSpanCondition.LessThan, timeSpan);
         }
+
+        /// <summary>
+        /// Asserts that the current <see cref="DateTime"/> has the <paramref name="expected"/> date.
+        /// </summary>
+        /// <param name="expected">The expected date portion of the current value.</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="reasonArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public AndConstraint<DateTimeAssertions> BeSameDateAs(DateTime expected, string because = "",
+            params object[] reasonArgs)
+        {
+            bool success = Execute.Assertion
+                .ForCondition(Subject.HasValue)
+                .BecauseOf(because, reasonArgs)
+                .FailWith("Expected a {context:date and time} with date {0}{reason}, but found a <null> DateTime.", expected);
+
+            if (success)
```

Comment:
You can avoid the need of a separate `success` variable by chaining assertions like https://github.com/dennisdoomen/fluentassertions/blob/master/Src/Core/Collections/CollectionAssertions.cs#L37